### PR TITLE
Clear out RSA decryption array-returning residuals.

### PR DESCRIPTION
**Pull request by bartonjs**
Initial draft has a non-resource exception and some commented out code to see if the UAP failure from the last PR that had this is repeatable so we can get exception diagnostics from it.

### CODE REVIEW

**Comment by bartonjs** [comment]
Ah, this is hitting the recent Windows bug where decrypt returns success when it meant more data accidentally. /me ponders.

File: src/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs [hunk 1]
```diff
@@ -260,7 +260,13 @@ private unsafe byte[] EncryptOrDecrypt(SafeNCryptKeyHandle key, ReadOnlySpan<byt
                 throw errorCode.ToCryptographicException();
             }
 
-            Array.Resize(ref output, numBytesNeeded);
+            if (numBytesNeeded != output.Length)
+            {
+                byte[] ret = output.AsSpan(0, numBytesNeeded).ToArray();
+                CryptographicOperations.ZeroMemory(output);
```
Comment (krwq) [inline src/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs]: should similar clearing happen in the code paths above? (i.e. in the branch on line 242 just before the new allocation)

Comment (bartonjs) [reply src/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs]: Yeah, it wouldn't hurt.  (It'd be definite yes if we knew that the underlying implementation was doing a partial write, so let's just do it as a defensive thing)

File: src/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs [hunk 2]
```diff
@@ -300,9 +307,17 @@ private unsafe bool TryEncryptOrDecrypt(SafeNCryptKeyHandle key, ReadOnlySpan<by
...
             out int bytesNeeded)
         {
-            return encrypt ?
+            ErrorCode errorCode = encrypt ?
                 Interop.NCrypt.NCryptEncrypt(key, input, input.Length, paddingInfo, output, output.Length, out bytesNeeded, paddingMode) :
                 Interop.NCrypt.NCryptDecrypt(key, input, input.Length, paddingInfo, output, output.Length, out bytesNeeded, paddingMode);
+
+            // Windows 10.1903 can return success when it meant NTE_BUFFER_TOO_SMALL.
+            if (errorCode == ErrorCode.ERROR_SUCCESS && bytesNeeded > output.Length)
+            {
```
Comment (krwq) [inline src/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs]: Wondering if we should clear here instead of per callsite

**Review by krwq — APPROVED** [review]
LGTM, consider changing per suggestion

File: src/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs (hunk 2 above)
Comment (bartonjs) [reply src/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs]: In the Span one it's up to the owner to manage their data.  Only the one where we're creating arrays for you are we responsible for clearing the arrays that are being abandoned to the GC.

**Pull request merged**